x/upgrade/internal/keeper/keeper.go

```diff
@@ -121,3 +125,8 @@ func (k Keeper) ApplyUpgrade(ctx sdk.Context, plan types.Plan) {
 	k.ClearUpgradePlan(ctx)
 	k.setDone(ctx, plan.Name)
 }
+
+// GetSkipUpgradeHeights returns the value of keeper's skipUpgradeHeights
+func (k Keeper) GetSkipUpgradeHeights() map[int64]bool {
```

Comment:
replace to:

```go
func (k Keeper) GetSkipUpgradeHeights() map[int64]bool {
    skipUpgradeHeights := make(map[int64]bool, len(k.skipUpgradeHeights))
    for k, v := range k.skipUpgradeHeights {
        skipUpgradeHeights[k] = v
    }

    return skipUpgradeHeights
}
```

Reply:
alternatively, you can return a sorted array

Reply:
Previously we were using arrays but with maps the code looks clean and easy to search.

Reply:
> replace to:
> 
> ```go
> func (k Keeper) GetSkipUpgradeHeights() map[int64]bool {
>     skipUpgradeHeights := make(map[int64]bool, len(k.skipUpgradeHeights))
>     for k, v := range k.skipUpgradeHeights {
>         skipUpgradeHeights[k] = v
>     }
> 
>     return skipUpgradeHeights
> }
> ```

I didn't really understand why we need this change? Am I missing something here?

Reply:
@fedekunze has a good point that you do not want to expose the mutable map here, or a called may change internal state. His suggestion is to do a standard copy of the map and return a copy. This is probably the simplest way to enforce immutability here, and works well with a small number of items.

Another solution in my mind would be to only expose the check and never the map itself. Like `func (k Keeper) IsSkipHeight(h int64) bool`. This can be used here https://github.com/cosmos/cosmos-sdk/pull/5367/files#diff-886a9b8a39f2bf5f0260a0f50ae14bafR28

Reply:
Yeah, I'd prefer `IsSkipHeight` instead of returning a map (copy).

Reply:
Updated, this looks very clean now. Thank you!

Reply:
Updated, this looks very clean now. Thank you!